app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/MoveFiles.java

```diff
@@ -68,12 +75,23 @@ public MoveFiles(ArrayList<ArrayList<HybridFileParcelable>> files, MainFragment
         this.mode = mode;
     }
 
+    @Override
+    public void onProgressUpdate(String... message) {
+        Toast.makeText(context, message[0], Toast.LENGTH_LONG).show();
+    }
+
     @Override
     protected Boolean doInBackground(ArrayList<String>... strings) {
         paths = strings[0];
 
         if (files.size() == 0) return true;
 
+        for (ArrayList<HybridFileParcelable> filesCurrent : files) {
```

Comment:
I would move this (new code) to right after the switch statement (after all other checks have been made), because of asymptotic cost.

Reply:
Moving them after switch can leave some variables uninitialized in some cases.

Reply:
`totalBytes` and `destinationSize` both seem to only be used after the `doInBackground()`.

Reply:
Yes, but for some cases in switch, we return out of ````doInBackground```` which can leave them uninitialised.